singer-io/tap-crossbeam
# Singer merge

# Description of change

This is a pretty large reworking of the tap that have been in progress for several weeks now. At the end of the day, the changes boil down to removing any reports-specific syncing and instead syncing all overlaps from Crossbeam. The `reports_data` stream is removed and in its place, these streams exist:

- `account`
- `lead`
- `user`
- `partner_account`
- `partner_lead`
- `partner_user`

Discovery for these streams is done ahead of time so that we know the final structure of the table and can enable catalog selection, however, there's one remaining TODO item to respect the selected fields within the sync process. 

# Manual QA steps
 
- I have run this for a test Crossbeam account as well as a real production account and synced to BigQuery using a third party BigQuery target. I have done a decent amount of verification that the data is correct, like comparing overlap counts to what we see within the Crossbeam product and ensuring that join keys are set up properly.
 
# Risks

- This is still a nascent tap, so there may be edge cases that are not covered.
- The fact that we don't yet respect field selection means that new fields can be sent to stdout that are not in the destination. However, as long as the targets are handling changes within the SCHEMA, this will be fine since any new fields are output to the schema.
 
# Rollback steps

 - revert this branch
